ansible_base/authentication/authenticator_plugins/oidc.py

```diff
@@ -215,3 +217,37 @@ def extra_data(self, user, backend, response, *args, **kwargs):
 
     def get_user_groups(self, extra_groups=[]):
         return extra_groups
+
+    def public_key(self):
+        key = self.setting("PUBLIC_KEY")
+        return (
+            "\n".join(
+                [
+                    "-----BEGIN PUBLIC KEY-----",
+                    key,
+                    "-----END PUBLIC KEY-----",
+                ]
+            )
+            if key
+            else None
+        )
+
+    def user_data(self, access_token, *args, **kwargs):
+        user_data = self.request(self.userinfo_url(), headers={"Authorization": f"Bearer {access_token}"})
+        if user_data.headers["Content-Type"] == "application/jwt":
+            pk = self.public_key()
+            if not pk:
+                logger.error(_("OIDC client sent encrypted user info response, but no public key found."))
+                return None
+            try:
+                data = jwt.decode(
+                    access_token,
+                    key=self.public_key(),
+                    algorithms=self.setting("JWT_ALGORITHMS"),
+                    audience=self.setting("KEY"),
+                )
+                return data
+            except PyJWTError as e:
```

Comment:
If we can't decrypt for some reason here should we just attempt to return `user_data.json()` just like the original method would?

Reply:
That will just throw the original json decode failed at char 1 exception, but fine with me if that is preferred.  That one will not trigger the exception handling middleware that would redirect back to login w/ a message (assuming that gets implemented) so that's why I returned the None.  But it could also be wrapped in one that would trigger the handler.  Or just raise AuthException here directly.

Reply:
I would be fine with either so long as a this code can still handle a not encrypted token.